patches/server/0844-Add-PlayerItemFrameChangeEvent.patch

```diff
@@ -0,0 +1,60 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: SamB440 <sam@islandearth.net>
+Date: Mon, 15 Nov 2021 18:10:10 +0000
+Subject: [PATCH] Add PlayerItemFrameChangeEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/decoration/ItemFrame.java b/src/main/java/net/minecraft/world/entity/decoration/ItemFrame.java
+index b829efdb40051a41b3bf1cabb8bf7d7c952797b5..631a1176868c83bf9e98fc62ec5f06459e90e42c 100644
+--- a/src/main/java/net/minecraft/world/entity/decoration/ItemFrame.java
++++ b/src/main/java/net/minecraft/world/entity/decoration/ItemFrame.java
+@@ -1,6 +1,7 @@
+ package net.minecraft.world.entity.decoration;
+ 
+ import javax.annotation.Nullable;
++import io.papermc.paper.event.player.PlayerItemFrameChangeEvent; // Paper
+ import net.minecraft.core.BlockPos;
+ import net.minecraft.core.Direction;
+ import net.minecraft.nbt.CompoundTag;
+@@ -185,6 +186,14 @@ public class ItemFrame extends HangingEntity {
+                     return true;
+                 }
+                 // CraftBukkit end
++                // Paper start - call PlayerItemFrameChangeEvent
++                if (source.getEntity() instanceof Player player) {
++                    var event = new PlayerItemFrameChangeEvent((org.bukkit.entity.Player) player.getBukkitEntity(), (org.bukkit.entity.ItemFrame) this.getBukkitEntity(), this.getItem().asBukkitCopy(), PlayerItemFrameChangeEvent.ItemFrameChangeAction.REMOVE);
```

Comment:
Any particular reason this isn't a fully qualified import similar to the `org.bukkit.entity.Player` in the same line ?

Reply:
The `org.bukkit.entity.Player` is mainly there because the nms player was already imported (which I use in the instanceof), so it needs that. 

As for the PlayerItemFrameChangeEvent and var, that was mostly to make it prettier and reduce the length of the line - I could change it but it looks really messy to me otherwise.

Reply:
yeah, I think its fine. looks like you are using it 6 times in this class o